src/components/invite/Invite.js

```diff
@@ -118,6 +124,123 @@ const ShareBox = ({ level }) => {
   )
 }
 
+const InputCodeBox = ({ navigateTo }) => {
+  const [code, setCode] = useState(userStorage.userProperties.get('inviterInviteCode') || '')
+  const inviteCodeUsed = userStorage.userProperties.get('inviterInviteCodeUsed')
+
+  //show component if reward not collected yet
+  const [visible, setVisible] = useState(!userStorage.userProperties.get('inviteBonusCollected'))
+  const [showDialog, hideDialog] = useDialog()
+  const extractedCode = get(extractQueryParams(code), 'inviteCode', code)
+  const isValidCode = extractedCode.length >= 10
+  const [disabled, setDisabled] = useState(!isValidCode) //disable button if code invalid or cant collect
+
```

Comment:
```suggestion

  const onUnableToCollect = useCallback(async () => {
     const isCitizen = await goodWallet.isCitizen()
       
    showDialog({
      image: <InfoIcon />,
      title: isCitizen ? 'Your inviter is not verified yet' : 'Claim your first G$s',
      message: isCitizen
        ? 'Ask your inviter to get verified by Claiming his first G$s'
        : 'In order to receive the reward',
      buttons: !isCitizen && [{
          text: 'Later',
          mode: 'text',
          color: theme.colors.gray80Percent,
          onPress: dismiss => {
            dismiss()
          },
        },{
          text: 'Claim Now',
          onPress: dismiss => {
            dismiss()
            navigateTo('Claim')
          },
        }],
      })
   }
  }, [navigateTo, showDialog])
  
```

Reply:
@sirpy  thanks for moving this out asi suggested because onSubmit fn become really huge